relm4-macros/src/util.rs

```diff
@@ -2,7 +2,53 @@ use proc_macro::TokenStream;
 use proc_macro2::Span as Span2;
 use syn::punctuated::Punctuated;
 
-use syn::{Ident, Path, PathArguments, PathSegment};
+use syn::spanned::Spanned;
+use syn::{Ident, ImplItem, ItemImpl, Path, PathArguments, PathSegment, Type, TypePath};
+
+pub(super) fn generate_widgets_type(
+    widgets_ty: Option<Type>,
```

Comment:
This can be determined from the item impl instead of passing it in.

Reply:
This is true, but then the `widgets_ty` field of the visitors is not needed anymore. What do you think? Should we just remove this from the visitor and handle it in this function instead?

Reply:
Good point. I don't feel strongly either way. Keeping it as-is is fine by me.